state-chain/runtime/src/runtime_apis.rs

```diff
@@ -55,6 +55,12 @@ pub struct RuntimeApiAccountInfoV2 {
 	pub restricted_balances: BTreeMap<EthereumAddress, u128>,
 }
 
+#[derive(Encode, Decode, Eq, PartialEq, TypeInfo, Serialize, Deserialize)]
+pub struct ChainAsset {
```

Comment:
I assume you need this type, as RpcAsset isn't no-std compatible? I believe we could remove it in the future, and have added a note to PRO-1157 to do so.

Reply:
In theory, we could move it to the runtime, but it's a type created for the RPC and I don't think it belongs there. Importing it from the RPC to the runtime is not possible because the dependencies are already set the other way around. But in any case we need a type that reflects the chain and the asset also for the custom RPC of the state-chain to the outer world. Internally, we can cast from Asset to Chain, but that's not possible if we return it from RPC. I think the cause of this type just shows that we have a general problem with our type system. The whole thing breaks at the moment with introduce USDC (for e.x) on any other chain.